### ISSUE TITLE
Hitting Segmentation fault when running Vitest .

### ISSUE THREAD

**Issue by Christopher-Stevers** [vitest-dev/vitest#3143]
### Describe the bug

When I run vitest on without limiting the threads, the process will crash half way through with the following output:
```
Segmentation fault (core dumped)
error Command failed with exit code 139.
```

### Reproduction

https://github.com/OpenQDev/OpenQ-Frontend/tree/vite-staging
I can't really provide a minimal reproduction because it seems to depend on the number of tests you run. To reproduce it with this repo just run `yarn` and then `yarn vitest`

### System Info

```shell
System:
    OS: Linux 5.4 Ubuntu 20.04.5 LTS (Focal Fossa)
    CPU: (16) x64 Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
    Memory: 26.75 GB / 31.36 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 19.7.0 - ~/nvm/current/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 9.5.0 - ~/nvm/current/bin/npm
  npmPackages:
    @vitejs/plugin-react: ^3.1.0 => 3.1.0 
    vitest: ^0.29.8 => 0.29.8
```

### Used Package Manager

yarn

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by rushil23** [comment]
Facing the same here. Really blocking our build flows. 

In our case we're facing this issue when running the tests on a github action

**Comment by Christopher-Stevers** [comment]
My work around at the moment has been to run it with the --no-threads flag when in circleci, but that slows down the workflow a lot.
Locally it works if for us if we are just testing a portion of the code base as long as we limit the threads to 1 or 2.

**Comment by nakleiderer** [comment]
Potentially similar root cause/duplicate of https://github.com/vitest-dev/vitest/issues/3143

Do you have a promise that sometimes rejects in your tests, similar to the reproduction on the linked issue?

**Comment by adrianisk** [comment]
We're potentially hitting this in Github Actions as well, works fine locally though... @nakleiderer did you mean to link to a different issue?

**Comment by nakleiderer** [comment]
@adrianisk Yes, apologies, I've updated the original comment. This is the link I meant to use: https://github.com/vitest-dev/vitest/issues/3106

**Comment by mortkarlberg** [comment]
We have the same issue. Works fine locally, but in CI on GitHub Actions we have to run sequentially with the --no-threads flag.

**Comment by adrianisk** [comment]
We're also hitting this issue which happened when we introduced an @aws-sdk dependency. I happened to stumble upon https://github.com/awslabs/aws-crt-nodejs/issues/452 while looking into something else, might be related?

**Comment by berenddeboer** [comment]
Same here.

**Comment by berenddeboer** [comment]
Can be fixed by setting `threads` in `vite.config.ts` to false, until a better solution is found.

**Comment by AriPerkkio** [comment]
Instead of disabling `threads` option completely, you could also move only the `@aws-sdk` related test cases to `node:child_process`. This way all other tests are running in `node:worker_threads` fast as usual, and the ones that are not compatible with `node:worker_threads` are run in `node:child_process`. 

Use the [`poolMatchGlobs` option](https://vitest.dev/config/#poolmatchglobs):

```ts
import { defineConfig } from 'vitest/config'

export default defineConfig({
  test: {
    poolMatchGlobs: [
      // Run AWS tests in child_process - @aws-sdk crashes when run in 'node:worker_threads'
      ['**/tests/aws-infra-etc-pattern/**', 'child_process'],
    ]
  }
})
```

**Comment by berenddeboer** [comment]
On repeatability, I can repeat this in a docker container with just 1 test.And indeed using @aws-sdk.

**Comment by AriPerkkio** [comment]
Fixed by https://github.com/vitest-dev/vitest/pull/3925 and https://github.com/vitest-dev/vitest/pull/4172.

Vitest 1.0.0-beta has now `--pool=forks` which uses multiple `node:child_process` in parallel. The previous `--no-threads` was using a single `node:child_process`.

If the code you are testing is incompatible with `node:worker_threads`, switch to `--pool=forks`. If you run into same error with that pool, feel free to open new issue with minimal reproduction.
